# Exploratory Data Analysis in R for Absolute Beginners

Note: You can consult the solution for this workspace in the file browser (`notebook-solution.ipynb`).

In this code-along, we'll learn the basics of R, the Tidyverse package, and the Lubridate package to determine which user onboarding 
flow the product team at DataCamp should focus on this quarter.

We will be using a synthetic dataset titled "user_page_view_history.csv" avaliable in our Workspace today.

This data set contains:
- Page view history of users.
- List of pages visited, the page that referred them, onboarding flow label, and date stamp of the visitation. 

## Table Of Contents
1. The Foundations
2. Exploring the Data
3. Onboarding Flows

## The Foundations of R

### What is R?

R is a programming language we can use to tell computers what to do! R can be used to solve problems as complex as prototyping a dashboard to as simple as 2 + 2.

In [47]:
# We could use R like a calculator
2 + 2

[1] 4

### What is a variable?

Sometimes after we have performed a calculation, we would like to save the output. We can do this by storing outputs as variables!

You can think of a variable as a box. Much like the boxes we use on a day to day basis, **variables** can store objects and assigned an alias for later. In R we store data objects inside variables using this pointer symbol "<-".

Here are three common data objects we like to store:
- **Integers** (`thenumberfour <- 4`)
- **Strings** (`thewordhello <- "hello"`)
- **Results Of Calculations** (`resultoftwoplustwo <- 2+2`)

In [48]:
# Here are three examples
resultoftwoplustwo <- 2 + 2

# We can view the contents of a variable by typing its name
resultoftwoplustwo

[1] 4

### What is a function?

![functioninmath](functioninmath.jpg)

Similar to functions in mathematics, **functions** take some input and create some output. In computer science you can create custom functions, although we won't be covering that in this lesson, or use premade function avaliable to us from the R. For example, we could write a function that computes 2 + 2 or we could use the function sum(1,1).

In [49]:
# Here is a simple and commonly used function
x <- sum(2,2)
x

[1] 4

Today we will be using these twelve functions:
- [library()](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/library)
- [read_csv()](https://www.rdocumentation.org/packages/qtl2/versions/0.24/topics/read_csv)
- [select()](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/select)
- [filter()](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/filter)
- [mutate()](https://www.rdocumentation.org/packages/dplyr/versions/0.5.0/topics/mutate)
- [floor_date()](https://www.rdocumentation.org/packages/lubridate/versions/1.3.3/topics/floor_date)
- [group_by()](https://www.rdocumentation.org/packages/dplyr/versions/1.0.10/topics/group_by)
- [summarise()](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/summarise)
- [n()](www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/n)
- [mean()](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/mean)
- [ungroup()](https://www.rdocumentation.org/packages/multiplyr/versions/0.1.1/topics/ungroup)
- [count()](https://www.rdocumentation.org/packages/plyr/versions/1.8.8/topics/count)

Lets try using the `library()` function to load the `tidyverse` package.

In [50]:
# Load tidyverse
library(tidyverse)

### What is a package?

The sum function we used earlier come a preloaded package when R is activated. For data analysis, we will be using the `tidyverse` package. A package containing many functions for manipulating data.

Now that we know the basics, lets get started!

## Exploratory Analysis

In order to figure out what onboarding flow we should optimze we have to figure out where users are coming from, where they are going, and when they are doing it. To do this we need to familiarize ourselves with the data we have avaliable. We typically do this by conducting exploratory analysis!

### Loading the Data

First things first, lets load our data.

#### Instructions

The dataset is stored in a CSV file named `user_page_view_history.csv` in the `data` folder.
- Load the data from the CSV file "user_page_view_history.csv" using the `read_csv` function and store the dataset to the variable `user_page_view_history`.
- View the data by typing out `user_page_view_history`.

In [51]:
# Read the CSV file "user_page_view_history.csv" and assign it to the user_page_view_history variable
user_page_view_history <- read_csv('data/user_page_view_history.csv')

# Type the variable name to view the result!
user_page_view_history

Rows: 40000 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (3): current_page_url, referral_page_url, user_flow
date (1): date_visited

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


current_page_url              
1     www.datacamp.com/courses      
2     www.datacamp.com/users/sign_up
3     www.datacamp.com/courses      
4     www.datacamp.com/courses      
5     www.datacamp.com/courses      
6     www.datacamp.com/users/sign_up
7     www.datacamp.com/courses      
8     www.datacamp.com/users/sign_up
9     www.datacamp.com/users/sign_up
10    www.datacamp.com/users/sign_up
11    www.datacamp.com/courses      
12    www.datacamp.com/courses      
13    www.datacamp.com/users/sign_up
14    www.datacamp.com/users/sign_up
15    www.datacamp.com/courses      
16    www.datacamp.com/users/sign_up
17    www.datacamp.com/courses      
18    www.datacamp.com/users/sign_up
19    www.datacamp.com/courses      
20    www.datacamp.com/courses      
21    www.datacamp.com/courses      
22    www.datacamp.com/courses      
23    www.datacamp.com/courses      
24    www.datacamp.com/users/sign_up
25    www.datacamp.com/users/sign_up
26    www.datacamp.com/users/sign_up
27    www.datacamp.com/users/sign_up
28    www.datacamp.com/users/sign_up
29    www.datacamp.com/users/sign_up
30    www.datacamp.com/courses      
⋮     ⋮                             
39971 www.datacamp.com/courses      
39972 www.datacamp.com/users/sign_up
39973 www.datacamp.com/users/sign_up
39974 www.datacamp.com/courses      
39975 www.datacamp.com/courses      
39976 www.datacamp.com/courses      
39977 www.datacamp.com/users/sign_up
39978 www.datacamp.com/courses      
39979 www.datacamp.com/courses      
39980 www.datacamp.com/courses      
39981 www.datacamp.com/courses      
39982 www.datacamp.com/courses      
39983 www.datacamp.com/users/sign_up
39984 www.datacamp.com/courses      
39985 www.datacamp.com/courses      
39986 www.datacamp.com/users/sign_up
39987 www.datacamp.com/users/sign_up
39988 www.datacamp.com/courses      
39989 www.datacamp.com/users/sign_up
39990 www.datacamp.com/users/sign_up
39991 www.datacamp.com/courses      
39992 www.datacamp.com/courses      
39993 www.datacamp.com/users/sign_up
39994 www.datacamp.com/courses      
39995 www.datacamp.com/courses      
39996 www.datacamp.com/users/sign_up
39997 www.datacamp.com/users/sign_up
39998 www.datacamp.com/users/sign_up
39999 www.datacamp.com/users/sign_up
40000 www.datacamp.com/users/sign_up
      referral_page_url                                        user_flow       
1     www.datacamp.com/blog/data-analytics-projects-all-levels blogToCourse    
2     www.datacamp.com/blog/data-analytics-projects-all-levels blogToSignup    
3     www.datacamp.com/blog/data-analytics-projects-all-levels blogToCourse    
4     www.datacamp.com/blog/data-analytics-projects-all-levels blogToCourse    
5     www.datacamp.com/blog/data-analytics-projects-all-levels blogToCourse    
6     www.datacamp.com/tutorial/create-histogram-plotly        tutorialToSignup
7     www.datacamp.com/blog/data-analytics-projects-all-levels blogToCourse    
8     www.datacamp.com/tutorial/create-histogram-plotly        tutorialToSignup
9     www.datacamp.com/blog/data-analytics-projects-all-levels blogToSignup    
10    www.datacamp.com/blog/data-analytics-projects-all-levels blogToSignup    
11    www.datacamp.com/tutorial/create-histogram-plotly        tutorialToCourse
12    www.datacamp.com/tutorial/create-histogram-plotly        tutorialToCourse
13    www.datacamp.com/blog/data-analytics-projects-all-levels blogToSignup    
14    www.datacamp.com/blog/data-analytics-projects-all-levels blogToSignup    
15    www.datacamp.com/tutorial/create-histogram-plotly        tutorialToCourse
16    www.datacamp.com/blog/data-analytics-projects-all-levels blogToSignup    
17    www.datacamp.com/tutorial/create-histogram-plotly        tutorialToCourse
18    www.datacamp.com/blog/data-analytics-projects-all-levels blogToSignup    
19    www.datacamp.com/blog/data-analytics-projects-all-levels blogToCourse    
20    www.datacamp.com/tutorial/create-histogram-plotly        tutorialToCourse
21    www.datacamp.com/blog/data

This looks a bit difficult to read, lets reorganise below using the `select()` function

In [52]:
# Reorganise the columns using select
user_page_view_history %>%
	select(date_visited, user_flow, current_page_url, referral_page_url)

date_visited user_flow        current_page_url              
1     2022-08-04   blogToCourse     www.datacamp.com/courses      
2     2022-05-17   blogToSignup     www.datacamp.com/users/sign_up
3     2022-11-02   blogToCourse     www.datacamp.com/courses      
4     2022-05-06   blogToCourse     www.datacamp.com/courses      
5     2022-10-26   blogToCourse     www.datacamp.com/courses      
6     2022-01-19   tutorialToSignup www.datacamp.com/users/sign_up
7     2022-02-10   blogToCourse     www.datacamp.com/courses      
8     2022-11-03   tutorialToSignup www.datacamp.com/users/sign_up
9     2022-01-21   blogToSignup     www.datacamp.com/users/sign_up
10    2022-12-12   blogToSignup     www.datacamp.com/users/sign_up
11    2022-09-12   tutorialToCourse www.datacamp.com/courses      
12    2022-10-08   tutorialToCourse www.datacamp.com/courses      
13    2022-05-14   blogToSignup     www.datacamp.com/users/sign_up
14    2022-03-24   blogToSignup     www.datacamp.com/users/sign_up
15    2022-07-18   tutorialToCourse www.datacamp.com/courses      
16    2022-12-16   blogToSignup     www.datacamp.com/users/sign_up
17    2022-02-20   tutorialToCourse www.datacamp.com/courses      
18    2022-02-13   blogToSignup     www.datacamp.com/users/sign_up
19    2022-04-17   blogToCourse     www.datacamp.com/courses      
20    2022-09-07   tutorialToCourse www.datacamp.com/courses      
21    2022-10-07   blogToCourse     www.datacamp.com/courses      
22    2022-11-12   blogToCourse     www.datacamp.com/courses      
23    2022-06-29   blogToCourse     www.datacamp.com/courses      
24    2022-03-20   tutorialToSignup www.datacamp.com/users/sign_up
25    2022-11-17   tutorialToSignup www.datacamp.com/users/sign_up
26    2022-05-10   blogToSignup     www.datacamp.com/users/sign_up
27    2022-08-19   tutorialToSignup www.datacamp.com/users/sign_up
28    2022-06-24   blogToSignup     www.datacamp.com/users/sign_up
29    2022-06-04   tutorialToSignup www.datacamp.com/users/sign_up
30    2022-05-08   blogToCourse     www.datacamp.com/courses      
⋮     ⋮            ⋮                ⋮                             
39971 2022-11-24   blogToCourse     www.datacamp.com/courses      
39972 2022-08-21   tutorialToSignup www.datacamp.com/users/sign_up
39973 2022-10-04   tutorialToSignup www.datacamp.com/users/sign_up
39974 2022-04-14   blogToCourse     www.datacamp.com/courses      
39975 2022-02-14   blogToCourse     www.datacamp.com/courses      
39976 2023-01-07   blogToCourse     www.datacamp.com/courses      
39977 2022-05-29   blogToSignup     www.datacamp.com/users/sign_up
39978 2022-06-11   tutorialToCourse www.datacamp.com/courses      
39979 2022-09-01   blogToCourse     www.datacamp.com/courses      
39980 2022-03-03   tutorialToCourse www.datacamp.com/courses      
39981 2022-05-28   blogToCourse     www.datacamp.com/courses      
39982 2022-09-18   blogToCourse     www.datacamp.com/courses      
39983 2022-02-14   tutorialToSignup www.datacamp.com/users/sign_up
39984 2022-11-26   blogToCourse     www.datacamp.com/courses      
39985 2022-04-04   tutorialToCourse www.datacamp.com/courses      
39986 2022-10-31   tutorialToSignup www.datacamp.com/users/sign_up
39987 2022-07-24   tutorialToSignup www.datacamp.com/users/sign_up
39988 2022-02-28   tutorialToCourse www.datacamp.com/courses      
39989 2022-06-20   tutorialToSignup www.datacamp.com/users/sign_up
39990 2022-04-30   tutorialToSignup www.datacamp.com/users/sign_up
39991 2022-10-02   blogToCourse     www.datacamp.com/courses      
39992 2022-01-21   blogToCourse     www.datacamp.com/courses      
39993 2022-03-13   tutorialToSignup www.datacamp.com/users/sign_up
39994 2022-11-28   tutorialToCourse www.datacamp.com/courses      
39995 2022-09-18   blogToCourse     www.datacamp.com/courses      
39996 2022-04-13   blogToSignup     www.datacamp.com/users/sign_up
39997 2022-11-21   blogToSignup     www.datacamp.com/users/sign_up
39998 2022-05-19   blogToSignup     www.datacamp.com/

#### Thoughts

After loading the data set their are a few key observations to note. We have a dataset titled user_page_view_history containing four columns:
- `current_page_url`: The URL of the current webpage.
- `referral_page_url`: The URL of the previous webpage.
- `user_flow`: Category for each onboarding flow.
- `date_visited`: The date of the page view.
We have 40,000 rows where each row contains the page viewed, the page that lead them there, an onboarding flow label, and the date the page view happened. 

### What pages do users enter DataCamp from?

Let's start by figuring out what pages users enter DataCamp from and how many users enter through each page!

#### Instructions

1. Use the `group_by()` function to group by the referral page in the `referral_page_url` column.
2. Use the `summarise()` and `n()` function to count the number of rows for each day.
3. Use a line plot to better visualize your data.

In [53]:
# Add code below!
user_page_view_history %>%
	group_by(referral_page_url) %>%
	summarise(page_visits = n()) %>%
	ungroup()

referral_page_url                                        page_visits
1 www.datacamp.com/blog/data-analytics-projects-all-levels 19891      
2 www.datacamp.com/tutorial/create-histogram-plotly        20109

This can also be simplified using the count() function.

In [54]:
# Add code below! Simplify using count()
user_page_view_history %>%
	count(referral_page_url)

referral_page_url                                        n    
1 www.datacamp.com/blog/data-analytics-projects-all-levels 19891
2 www.datacamp.com/tutorial/create-histogram-plotly        20109

#### Thoughts

Take notes on your findings below by hovering over the cell below and clicking Add Text!

It looks like users enter our website through two different webpages:
- [A Blog](https://www.datacamp.com/blog/data-analytics-projects-all-levels)
- [A Tutorial](https://www.datacamp.com/tutorial/create-histogram-plotly)
The blog page appears to be more users onto our website, but only by a thin margin.

Two different ways users enter datacamp, blog or a tutorial. Tutorials have bit more traffic, but by a very thin margin.

### What do they do after?

Now you try! What pages do users visit after they land? How many users entered through each page?

#### Instructions

Add your code below, visualize your results, and take notes on your findings!

In [55]:
# Add code below! Try solving using the group_by() function.
user_page_view_history %>%
	group_by(current_page_url) %>%
	summarise(page_visits = n()) %>%
	ungroup()

current_page_url               page_visits
1 www.datacamp.com/courses       19952      
2 www.datacamp.com/users/sign_up 20048

#### Thoughts

Take notes on your findings below!

It looks like users enter our website through two different webpages:
- [Courses](https://www.datacamp.com/courses)
- [Registration](www.datacamp.com/users/sign_up)
The course page appears to be the most popular next step for users on our website, but only by a thin margin.

Two pages users go after, courses page or the sign up page with equal traffic. 

### What paths do users take through onboarding?

#### Instructions

Add your code below, visualize your results, and take notes on your findings!

In [56]:
# Add code below! Try solving using the count() function. 
user_page_view_history %>%
	count(user_flow) %>%
	ungroup()

user_flow        n    
1 blogToCourse      9948
2 blogToSignup      9943
3 tutorialToCourse 10004
4 tutorialToSignup 10105

#### Thoughts

Take notes on your findings below!


There are four flows users tend to use when onboarding. This makes sense because there are two referral page types and two current pages types. Although there is a bit of variability, all three onboarding flows seem like equal candidates. 

At this point it looks like we can choose any path. How can we make a better decision?

### When did these visits happen?

We have one more column we haven't explored yet, the time column! It's important to know how old the data we are working with is and what time frame we are working with. Lets visualize the number of page visits over time!

#### Instructions

Graph the number of visits over time! If the graph is difficult to read, group by the week instead of the day.

In [57]:
library(lubridate)

user_page_view_history %>%
	filter(date_visited <= "2022-12-31") %>%
	mutate(week = floor_date(date_visited, "week")) %>%
	group_by(week) %>%
	summarise(page_visits = n()) %>%
	ungroup()

week       page_visits
1  2022-01-09 699        
2  2022-01-16 863        
3  2022-01-23 808        
4  2022-01-30 813        
5  2022-02-06 824        
6  2022-02-13 804        
7  2022-02-20 784        
8  2022-02-27 826        
9  2022-03-06 833        
10 2022-03-13 818        
11 2022-03-20 815        
12 2022-03-27 843        
13 2022-04-03 831        
14 2022-04-10 861        
15 2022-04-17 805        
16 2022-04-24 799        
17 2022-05-01 821        
18 2022-05-08 859        
19 2022-05-15 846        
20 2022-05-22 861        
21 2022-05-29 812        
22 2022-06-05 791        
23 2022-06-12 773        
24 2022-06-19 793        
25 2022-06-26 811        
26 2022-07-03 796        
27 2022-07-10 697        
28 2022-07-17 709        
29 2022-07-24 689        
30 2022-07-31 751        
31 2022-08-07 759        
32 2022-08-14 716        
33 2022-08-21 700        
34 2022-08-28 691        
35 2022-09-04 678        
36 2022-09-11 749        
37 2022-09-18 677        
38 2022-09-25 751        
39 2022-10-02 678        
40 2022-10-09 668        
41 2022-10-16 665        
42 2022-10-23 713        
43 2022-10-30 677        
44 2022-11-06 677        
45 2022-11-13 714        
46 2022-11-20 754        
47 2022-11-27 740        
48 2022-12-04 718        
49 2022-12-11 737        
50 2022-12-18 746        
51 2022-12-25 730

#### Thoughts

Take notes on your findings below!

There are two major take aways from this plot. The first observation is that our data goes as far back as Janurary 1st 2022. The second is that during the month of July 2022 there was a drop in traffic! This is our first major find! Up until now path traffic appeared relatively consistant, but it seems we used to average around 120 visits per day, but during the month of July, 2022 it dropped. This graph opens the door to a number of questions!
- What was most visits we've had in a day?
- What day did we have the least? Did something happen?
- What is the average?
- What happened between July 2022 and August 2022?

#### Instructions

Quantify how large the drop in visitations was using the formula:

(new average - old average) / old average

In [58]:
page_views_weekly <- user_page_view_history %>%
	filter(date_visited <= "2022-12-31") %>%
	mutate(week = floor_date(date_visited, "week")) 

page_views_weekly %>%
	group_by(week) %>%
	summarise(page_visits = n()) %>%
	ungroup() %>%
	filter(week >= "2022-08-01") %>%
	mutate(average_views = mean(page_visits))

week       page_visits average_views
1  2022-08-07 759         711.3333     
2  2022-08-14 716         711.3333     
3  2022-08-21 700         711.3333     
4  2022-08-28 691         711.3333     
5  2022-09-04 678         711.3333     
6  2022-09-11 749         711.3333     
7  2022-09-18 677         711.3333     
8  2022-09-25 751         711.3333     
9  2022-10-02 678         711.3333     
10 2022-10-09 668         711.3333     
11 2022-10-16 665         711.3333     
12 2022-10-23 713         711.3333     
13 2022-10-30 677         711.3333     
14 2022-11-06 677         711.3333     
15 2022-11-13 714         711.3333     
16 2022-11-20 754         711.3333     
17 2022-11-27 740         711.3333     
18 2022-12-04 718         711.3333     
19 2022-12-11 737         711.3333     
20 2022-12-18 746         711.3333     
21 2022-12-25 730         711.3333

#### Thoughts

Take notes on your findings below!

Before August 1st, 2022 we averaged 801 viewed per week, now we average 711. This is an 11% decrease in traffic!

### How has the landing page performed over time?

Now that we have a better handle on our data, lets figure out which onboarding flow to focus on. In order to decide between the flows, lets see how each landing page is performing over time!

In [59]:
# Add code below!
#page_views_weekly %>%
	#group_by(week, current_page_url) %>%
	#summarise(page_visits = n()) %>%
	#ungroup()

user_page_view_history %>%
	count(date_visited)


date_visited n  
1   2022-01-10   122
2   2022-01-11   122
3   2022-01-12   112
4   2022-01-13   114
5   2022-01-14    99
6   2022-01-15   130
7   2022-01-16   112
8   2022-01-17   125
9   2022-01-18   104
10  2022-01-19   123
11  2022-01-20   127
12  2022-01-21   153
13  2022-01-22   119
14  2022-01-23   105
15  2022-01-24   115
16  2022-01-25   134
17  2022-01-26   109
18  2022-01-27   109
19  2022-01-28   115
20  2022-01-29   121
21  2022-01-30   105
22  2022-01-31   121
23  2022-02-01   109
24  2022-02-02   115
25  2022-02-03   116
26  2022-02-04   123
27  2022-02-05   124
28  2022-02-06   108
29  2022-02-07   122
30  2022-02-08   123
⋮   ⋮            ⋮  
337 2022-12-12   111
338 2022-12-13   105
339 2022-12-14   109
340 2022-12-15   115
341 2022-12-16    82
342 2022-12-17   101
343 2022-12-18   106
344 2022-12-19   124
345 2022-12-20   107
346 2022-12-21    97
347 2022-12-22   102
348 2022-12-23    99
349 2022-12-24   111
350 2022-12-25   105
351 2022-12-26   115
352 2022-12-27   111
353 2022-12-28   103
354 2022-12-29    96
355 2022-12-30   106
356 2022-12-31    94
357 2023-01-01   120
358 2023-01-02    97
359 2023-01-03   114
360 2023-01-04    93
361 2023-01-05    97
362 2023-01-06   104
363 2023-01-07   101
364 2023-01-08   102
365 2023-01-09   109
366 2023-01-10    90

#### Thoughts

Take notes on your findings below!

When we group by the date visited and the referral page, we see that the tutorial page has retained consistant performance over this past year, but the blog page has suffered a major drop in traffic. This opens the door to multiple questions:

- Why hasn't the tutorial page grown over time? Is this an area we need to expand?
- What happened to the blog page in July? Could this be an engineering bug? 
- How did this effect the subsequent pages?  

When we perform the same exercise on our current page urls, we see a similar pattern on our registration pages. It would also appear that the majority of registration traffic was driven by our blog page. Depending on the percentage of users that purchase after registration this could be a serious cause for concern!


### Which onboarding flow should the product team focus on?

When we first began our analysis in aggregate, the data showed it did not matter which onboarding path we chose to optimze as all four had roughly the same traffic. Upon further investigation it would seem that the blog page and the registration page have seen large drop in traffic. Lets see if the drop in blog page views is the cause of the drop in registration page views.

#### Instructions

Now you try! Graph the user_flows over time and record your findings!

In [60]:
# Add code below!
# TODO
#page_views_weekly %>%
#	group_by(week, user_flow) %>%
#	summarise(page_visits = n()) %>%
#	ungroup()

user_page_view_history %>%
	mutate(week = floor_date(date_visited, "week")) %>%
	count(week, user_flow)

week       user_flow        n  
1   2022-01-09 blogToCourse     151
2   2022-01-09 blogToSignup     223
3   2022-01-09 tutorialToCourse 167
4   2022-01-09 tutorialToSignup 158
5   2022-01-16 blogToCourse     195
6   2022-01-16 blogToSignup     265
7   2022-01-16 tutorialToCourse 208
8   2022-01-16 tutorialToSignup 195
9   2022-01-23 blogToCourse     199
10  2022-01-23 blogToSignup     237
11  2022-01-23 tutorialToCourse 184
12  2022-01-23 tutorialToSignup 188
13  2022-01-30 blogToCourse     209
14  2022-01-30 blogToSignup     241
15  2022-01-30 tutorialToCourse 171
16  2022-01-30 tutorialToSignup 192
17  2022-02-06 blogToCourse     172
18  2022-02-06 blogToSignup     251
19  2022-02-06 tutorialToCourse 177
20  2022-02-06 tutorialToSignup 224
21  2022-02-13 blogToCourse     192
22  2022-02-13 blogToSignup     232
23  2022-02-13 tutorialToCourse 195
24  2022-02-13 tutorialToSignup 185
25  2022-02-20 blogToCourse     183
26  2022-02-20 blogToSignup     257
27  2022-02-20 tutorialToCourse 176
28  2022-02-20 tutorialToSignup 168
29  2022-02-27 blogToCourse     173
30  2022-02-27 blogToSignup     259
⋮   ⋮          ⋮                ⋮  
183 2022-11-20 tutorialToCourse 186
184 2022-11-20 tutorialToSignup 220
185 2022-11-27 blogToCourse     212
186 2022-11-27 blogToSignup     131
187 2022-11-27 tutorialToCourse 193
188 2022-11-27 tutorialToSignup 204
189 2022-12-04 blogToCourse     186
190 2022-12-04 blogToSignup     148
191 2022-12-04 tutorialToCourse 192
192 2022-12-04 tutorialToSignup 192
193 2022-12-11 blogToCourse     188
194 2022-12-11 blogToSignup     143
195 2022-12-11 tutorialToCourse 185
196 2022-12-11 tutorialToSignup 221
197 2022-12-18 blogToCourse     203
198 2022-12-18 blogToSignup     133
199 2022-12-18 tutorialToCourse 209
200 2022-12-18 tutorialToSignup 201
201 2022-12-25 blogToCourse     184
202 2022-12-25 blogToSignup     138
203 2022-12-25 tutorialToCourse 194
204 2022-12-25 tutorialToSignup 214
205 2023-01-01 blogToCourse     192
206 2023-01-01 blogToSignup     140
207 2023-01-01 tutorialToCourse 199
208 2023-01-01 tutorialToSignup 195
209 2023-01-08 blogToCourse      66
210 2023-01-08 blogToSignup      60
211 2023-01-08 tutorialToCourse  98
212 2023-01-08 tutorialToSignup  77

#### Instructions

Now you try! Calculate the average page views before and after the first of August!

In [61]:
# Add code below!
page_views_weekly %>%
	group_by(week, user_flow) %>%
	summarise(page_visits = n()) %>%
	ungroup() %>%
	filter(week >= "2022-08-01",
          user_flow == "blogToSignup") %>%
	mutate(average = mean(page_visits))

`summarise()` has grouped output by 'week'. You can override using the
`.groups` argument.


week       user_flow    page_visits average 
1  2022-08-07 blogToSignup 142         133.1905
2  2022-08-14 blogToSignup 142         133.1905
3  2022-08-21 blogToSignup 112         133.1905
4  2022-08-28 blogToSignup 138         133.1905
5  2022-09-04 blogToSignup 133         133.1905
6  2022-09-11 blogToSignup 133         133.1905
7  2022-09-18 blogToSignup 154         133.1905
8  2022-09-25 blogToSignup 139         133.1905
9  2022-10-02 blogToSignup 135         133.1905
10 2022-10-09 blogToSignup 119         133.1905
11 2022-10-16 blogToSignup 112         133.1905
12 2022-10-23 blogToSignup 114         133.1905
13 2022-10-30 blogToSignup 125         133.1905
14 2022-11-06 blogToSignup 131         133.1905
15 2022-11-13 blogToSignup 138         133.1905
16 2022-11-20 blogToSignup 137         133.1905
17 2022-11-27 blogToSignup 131         133.1905
18 2022-12-04 blogToSignup 148         133.1905
19 2022-12-11 blogToSignup 143         133.1905
20 2022-12-18 blogToSignup 133         133.1905
21 2022-12-25 blogToSignup 138         133.1905

#### Thoughts

Take notes on your findings below! 

The blog to sign up flow was the once the most popular onboarding and has seen as 42% drop since July 2022. This should be brought up to the product team.